lr-cli/cortx_setup/commands/storage/config.py

```diff
@@ -259,22 +345,53 @@ def run(self, **kwargs):
                     if self.enclosure_id:
                         self.update_pillar_and_conf('name', name)
                     else:
-                        self.logger.error("Please set 'user, password, primary ip and port' first")
-                        raise RuntimeError("Cannot set name before setting user, password, ip and port")
+                        self.logger.error("Enclosure ID is not set"
+                            "Run following command to set the enclosure id:"
+                            "cortx_setup storage config --user <user>"
+                            " --password <passwd> --ip <ip> --port <port>")
+                        raise RuntimeError(
+                            "Cannot set enclosure name without enclosure id"
+                        )
 
                 if storage_type is not None:
                     if self.enclosure_id:
-                        self.update_pillar_and_conf('storage_type', storage_type)
+                        if storage_type == "virtual":
+                            self.logger.error(
+                            "Storage type can not be 'virtual' for HW")
+                            raise ValueError("Incorrect argument value provided")
+                        else:
+                            self.update_pillar_and_conf(
+                                'storage_type',
+                                storage_type
+                            )
                     else:
-                        self.logger.error("Please set 'user, password, primary ip and port' first")
-                        raise RuntimeError("Cannot set storage_type before setting user, password, ip and port")
+                        self.logger.error("Enclosure ID is not set"
+                            "Run following command to set the enclosure id:"
+                            "cortx_setup storage config --user <user>"
+                            " --password <passwd> --ip <ip> --port <port>")
+                        raise RuntimeError(
+                            "Cannot set enclosure type without enclosure id"
+                        )
 
                 if controller_type is not None:
```

Comment:
Do we not have default controller_type for VM?

Reply:
No, we set it for type (enclosure type).
For VM only {name, type, cvg, data and metadata_devices} arguments are accepted. If user provides controller_type on VM then onlly "virtual" is accepted.